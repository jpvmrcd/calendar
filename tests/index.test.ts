import { Calendar, OnCellAddedArgs, OnHeaderCellAddedArgs } from '../src/index';

test('null element should throw error', () => {
  expect(() => new Calendar(null).render()).toThrow('element is null');
});

test('render without args should render current calendar', () => {
  jest.useFakeTimers('modern');
  jest.setSystemTime(new Date('12 May 2021 00:00:00 UTC'));

  const calendar = new Calendar(document.createElement('div'));
  calendar.render();
  expect(calendar.element.outerHTML).toMatchSnapshot();

  jest.useRealTimers();
});

test('render January 1900 should render correct calendar', () => {
  const calendar = new Calendar(document.createElement('div'));
  calendar.render(1900, 0);

  expect(calendar.element.outerHTML).toMatchSnapshot();
});

test('render December 2200 should render correct calendar', () => {
  const calendar = new Calendar(document.createElement('div'));
  calendar.render(2200, 11);

  expect(calendar.element.outerHTML).toMatchSnapshot();
});

test('setting options.dayNames should reflect on calendar', () => {
  const calendar = new Calendar(document.createElement('div'));
  calendar.options.dayNames = ['Su', 'Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa'];
  calendar.render(2021, 5);

  expect(calendar.element.outerHTML).toMatchSnapshot();
});

test('setting options.startDay that does not match any dayNames should throw an error', () => {
  const calendar = new Calendar(document.createElement('div'));
  calendar.options.startDay = 'Sunday';

  expect(() => calendar.render(2021, 5)).toThrow('startDay [Sunday] must be one of options.dayNames\'s values [Sun,Mon,Tue,Wed,Thu,Fri,Sat].');
});

test('setting options.startDay to Mon should render a calendar that starts on a Monday', () => {
  const calendar = new Calendar(document.createElement('div'));
  calendar.options.startDay = 'Mon';
  calendar.render(2021, 5);

  expect(calendar.element.outerHTML).toMatchSnapshot();
});

test('setting options.startDay to Tue should render a calendar that starts on a Tuesday', () => {
  const calendar = new Calendar(document.createElement('div'));
  calendar.options.startDay = 'Tue';
  calendar.render(2021, 5);

  expect(calendar.element.outerHTML).toMatchSnapshot();
});

test('setting options.startDay to Wed should render a calendar that starts on a Wednesday', () => {
  const calendar = new Calendar(document.createElement('div'));
  calendar.options.startDay = 'Wed';
  calendar.render(2021, 5);

  expect(calendar.element.outerHTML).toMatchSnapshot();
});

test('setting options.startDay to Thu should render a calendar that starts on a Thursday', () => {
  const calendar = new Calendar(document.createElement('div'));
  calendar.options.startDay = 'Thu';
  calendar.render(2021, 5);

  expect(calendar.element.outerHTML).toMatchSnapshot();
});

test('setting options.startDay to Fri should render a calendar that starts on a Friday', () => {
  const calendar = new Calendar(document.createElement('div'));
  calendar.options.startDay = 'Fri';
  calendar.render(2021, 5);

  expect(calendar.element.outerHTML).toMatchSnapshot();
});

test('setting options.startDay to Sat should render a calendar that starts on a Saturday', () => {
  const calendar = new Calendar(document.createElement('div'));
  calendar.options.startDay = 'Sat';
  calendar.render(2021, 5);

  expect(calendar.element.outerHTML).toMatchSnapshot();
});

test('setting options.startDay to Sun should render a calendar that starts on a Sunday', () => {
  const calendar = new Calendar(document.createElement('div'));
  calendar.options.startDay = 'Sun';
  calendar.render(2021, 5);

  expect(calendar.element.outerHTML).toMatchSnapshot();
});

test('calling render again should update the calendar rendered', () => {
  const calendar = new Calendar(document.createElement('div'));
  calendar.render(2021, 5);
  calendar.render(2021, 6);

  expect(calendar.element.outerHTML).toMatchSnapshot();
});

test('calling onHeaderCellAdded to update rendered cell should reflect on actual rendered calendar', () => {
  const calendar = new Calendar(document.createElement('div'));
  calendar.options.onHeaderCellAdded = (arg: OnHeaderCellAddedArgs) => {
    arg.th.innerHTML = `<div>${arg.day}</div>`;
  };
  calendar.render(2024, 10);

  expect(calendar.element.outerHTML).toMatchSnapshot();
});

test('setting onHeaderCellAdded to invalid type should render default calendar', () => {
  const calendar = new Calendar(document.createElement('div'));
  calendar.options.onHeaderCellAdded = 'test' as any;
  calendar.render(2024, 10);

  expect(calendar.element.outerHTML).toMatchSnapshot();
});

test('calling onCellAdded to update rendered cell should reflect on actual rendered calendar', () => {
  const calendar = new Calendar(document.createElement('div'));
  calendar.options.onCellAdded = (arg: OnCellAddedArgs) => {
    arg.td.innerHTML = `<div>${+arg.dateISOString.substring(8, 10)}</div>`;
  };
  calendar.render(2021, 5);

  expect(calendar.element.outerHTML).toMatchSnapshot();
});

test('setting onCellAdded to invalid type should render default calendar', () => {
  const calendar = new Calendar(document.createElement('div'));
  calendar.options.onCellAdded = 'test' as any;
  calendar.render(2021, 5);

  expect(calendar.element.outerHTML).toMatchSnapshot();
});

test('setting onDateClicked with a valid function should attach a click event to td with date', () => {
  const calendar = new Calendar(document.createElement('div'));
  const mockCallback = jest.fn((x) => x);
  calendar.options.onDateClicked = mockCallback;
  calendar.render(2021, 5);
  (calendar.element.querySelector('td[data-date="2021-06-01"]') as any).click();

  expect(mockCallback.mock.calls.length).toBe(1);
});
