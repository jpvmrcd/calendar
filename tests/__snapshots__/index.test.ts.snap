// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calling onCellAdded to update rendered cell should reflect on actual rendered calendar 1`] = `"<div><table><thead><tr><th>Sun</th><th>Mon</th><th>Tue</th><th>Wed</th><th>Thu</th><th>Fri</th><th>Sat</th></tr></thead><tbody><tr><td></td><td></td><td data-date="2021-06-01"><div>1</div></td><td data-date="2021-06-02"><div>2</div></td><td data-date="2021-06-03"><div>3</div></td><td data-date="2021-06-04"><div>4</div></td><td data-date="2021-06-05"><div>5</div></td></tr><tr><td data-date="2021-06-06"><div>6</div></td><td data-date="2021-06-07"><div>7</div></td><td data-date="2021-06-08"><div>8</div></td><td data-date="2021-06-09"><div>9</div></td><td data-date="2021-06-10"><div>10</div></td><td data-date="2021-06-11"><div>11</div></td><td data-date="2021-06-12"><div>12</div></td></tr><tr><td data-date="2021-06-13"><div>13</div></td><td data-date="2021-06-14"><div>14</div></td><td data-date="2021-06-15"><div>15</div></td><td data-date="2021-06-16"><div>16</div></td><td data-date="2021-06-17"><div>17</div></td><td data-date="2021-06-18"><div>18</div></td><td data-date="2021-06-19"><div>19</div></td></tr><tr><td data-date="2021-06-20"><div>20</div></td><td data-date="2021-06-21"><div>21</div></td><td data-date="2021-06-22"><div>22</div></td><td data-date="2021-06-23"><div>23</div></td><td data-date="2021-06-24"><div>24</div></td><td data-date="2021-06-25"><div>25</div></td><td data-date="2021-06-26"><div>26</div></td></tr><tr><td data-date="2021-06-27"><div>27</div></td><td data-date="2021-06-28"><div>28</div></td><td data-date="2021-06-29"><div>29</div></td><td data-date="2021-06-30"><div>30</div></td><td></td><td></td><td></td></tr></tbody></table></div>"`;

exports[`calling onHeaderCellAdded to update rendered cell should reflect on actual rendered calendar 1`] = `"<div><table><thead><tr><th><div>Sun</div></th><th><div>Mon</div></th><th><div>Tue</div></th><th><div>Wed</div></th><th><div>Thu</div></th><th><div>Fri</div></th><th><div>Sat</div></th></tr></thead><tbody><tr><td></td><td></td><td></td><td></td><td></td><td data-date="2024-11-01">1</td><td data-date="2024-11-02">2</td></tr><tr><td data-date="2024-11-03">3</td><td data-date="2024-11-04">4</td><td data-date="2024-11-05">5</td><td data-date="2024-11-06">6</td><td data-date="2024-11-07">7</td><td data-date="2024-11-08">8</td><td data-date="2024-11-09">9</td></tr><tr><td data-date="2024-11-10">10</td><td data-date="2024-11-11">11</td><td data-date="2024-11-12">12</td><td data-date="2024-11-13">13</td><td data-date="2024-11-14">14</td><td data-date="2024-11-15">15</td><td data-date="2024-11-16">16</td></tr><tr><td data-date="2024-11-17">17</td><td data-date="2024-11-18">18</td><td data-date="2024-11-19">19</td><td data-date="2024-11-20">20</td><td data-date="2024-11-21">21</td><td data-date="2024-11-22">22</td><td data-date="2024-11-23">23</td></tr><tr><td data-date="2024-11-24">24</td><td data-date="2024-11-25">25</td><td data-date="2024-11-26">26</td><td data-date="2024-11-27">27</td><td data-date="2024-11-28">28</td><td data-date="2024-11-29">29</td><td data-date="2024-11-30">30</td></tr></tbody></table></div>"`;

exports[`calling render again should update the calendar rendered 1`] = `"<div><table><thead><tr><th>Sun</th><th>Mon</th><th>Tue</th><th>Wed</th><th>Thu</th><th>Fri</th><th>Sat</th></tr></thead><tbody><tr><td></td><td></td><td></td><td></td><td data-date="2021-07-01">1</td><td data-date="2021-07-02">2</td><td data-date="2021-07-03">3</td></tr><tr><td data-date="2021-07-04">4</td><td data-date="2021-07-05">5</td><td data-date="2021-07-06">6</td><td data-date="2021-07-07">7</td><td data-date="2021-07-08">8</td><td data-date="2021-07-09">9</td><td data-date="2021-07-10">10</td></tr><tr><td data-date="2021-07-11">11</td><td data-date="2021-07-12">12</td><td data-date="2021-07-13">13</td><td data-date="2021-07-14">14</td><td data-date="2021-07-15">15</td><td data-date="2021-07-16">16</td><td data-date="2021-07-17">17</td></tr><tr><td data-date="2021-07-18">18</td><td data-date="2021-07-19">19</td><td data-date="2021-07-20">20</td><td data-date="2021-07-21">21</td><td data-date="2021-07-22">22</td><td data-date="2021-07-23">23</td><td data-date="2021-07-24">24</td></tr><tr><td data-date="2021-07-25">25</td><td data-date="2021-07-26">26</td><td data-date="2021-07-27">27</td><td data-date="2021-07-28">28</td><td data-date="2021-07-29">29</td><td data-date="2021-07-30">30</td><td data-date="2021-07-31">31</td></tr></tbody></table></div>"`;

exports[`render December 2200 should render correct calendar 1`] = `"<div><table><thead><tr><th>Sun</th><th>Mon</th><th>Tue</th><th>Wed</th><th>Thu</th><th>Fri</th><th>Sat</th></tr></thead><tbody><tr><td></td><td data-date="2200-12-01">1</td><td data-date="2200-12-02">2</td><td data-date="2200-12-03">3</td><td data-date="2200-12-04">4</td><td data-date="2200-12-05">5</td><td data-date="2200-12-06">6</td></tr><tr><td data-date="2200-12-07">7</td><td data-date="2200-12-08">8</td><td data-date="2200-12-09">9</td><td data-date="2200-12-10">10</td><td data-date="2200-12-11">11</td><td data-date="2200-12-12">12</td><td data-date="2200-12-13">13</td></tr><tr><td data-date="2200-12-14">14</td><td data-date="2200-12-15">15</td><td data-date="2200-12-16">16</td><td data-date="2200-12-17">17</td><td data-date="2200-12-18">18</td><td data-date="2200-12-19">19</td><td data-date="2200-12-20">20</td></tr><tr><td data-date="2200-12-21">21</td><td data-date="2200-12-22">22</td><td data-date="2200-12-23">23</td><td data-date="2200-12-24">24</td><td data-date="2200-12-25">25</td><td data-date="2200-12-26">26</td><td data-date="2200-12-27">27</td></tr><tr><td data-date="2200-12-28">28</td><td data-date="2200-12-29">29</td><td data-date="2200-12-30">30</td><td data-date="2200-12-31">31</td><td></td><td></td><td></td></tr></tbody></table></div>"`;

exports[`render January 1900 should render correct calendar 1`] = `"<div><table><thead><tr><th>Sun</th><th>Mon</th><th>Tue</th><th>Wed</th><th>Thu</th><th>Fri</th><th>Sat</th></tr></thead><tbody><tr><td></td><td data-date="1900-01-01">1</td><td data-date="1900-01-02">2</td><td data-date="1900-01-03">3</td><td data-date="1900-01-04">4</td><td data-date="1900-01-05">5</td><td data-date="1900-01-06">6</td></tr><tr><td data-date="1900-01-07">7</td><td data-date="1900-01-08">8</td><td data-date="1900-01-09">9</td><td data-date="1900-01-10">10</td><td data-date="1900-01-11">11</td><td data-date="1900-01-12">12</td><td data-date="1900-01-13">13</td></tr><tr><td data-date="1900-01-14">14</td><td data-date="1900-01-15">15</td><td data-date="1900-01-16">16</td><td data-date="1900-01-17">17</td><td data-date="1900-01-18">18</td><td data-date="1900-01-19">19</td><td data-date="1900-01-20">20</td></tr><tr><td data-date="1900-01-21">21</td><td data-date="1900-01-22">22</td><td data-date="1900-01-23">23</td><td data-date="1900-01-24">24</td><td data-date="1900-01-25">25</td><td data-date="1900-01-26">26</td><td data-date="1900-01-27">27</td></tr><tr><td data-date="1900-01-28">28</td><td data-date="1900-01-29">29</td><td data-date="1900-01-30">30</td><td data-date="1900-01-31">31</td><td></td><td></td><td></td></tr></tbody></table></div>"`;

exports[`render without args should render current calendar 1`] = `"<div><table><thead><tr><th>Sun</th><th>Mon</th><th>Tue</th><th>Wed</th><th>Thu</th><th>Fri</th><th>Sat</th></tr></thead><tbody><tr><td></td><td></td><td></td><td></td><td></td><td></td><td data-date="2021-05-01">1</td></tr><tr><td data-date="2021-05-02">2</td><td data-date="2021-05-03">3</td><td data-date="2021-05-04">4</td><td data-date="2021-05-05">5</td><td data-date="2021-05-06">6</td><td data-date="2021-05-07">7</td><td data-date="2021-05-08">8</td></tr><tr><td data-date="2021-05-09">9</td><td data-date="2021-05-10">10</td><td data-date="2021-05-11">11</td><td data-date="2021-05-12" class="today">12</td><td data-date="2021-05-13">13</td><td data-date="2021-05-14">14</td><td data-date="2021-05-15">15</td></tr><tr><td data-date="2021-05-16">16</td><td data-date="2021-05-17">17</td><td data-date="2021-05-18">18</td><td data-date="2021-05-19">19</td><td data-date="2021-05-20">20</td><td data-date="2021-05-21">21</td><td data-date="2021-05-22">22</td></tr><tr><td data-date="2021-05-23">23</td><td data-date="2021-05-24">24</td><td data-date="2021-05-25">25</td><td data-date="2021-05-26">26</td><td data-date="2021-05-27">27</td><td data-date="2021-05-28">28</td><td data-date="2021-05-29">29</td></tr><tr><td data-date="2021-05-30">30</td><td data-date="2021-05-31">31</td><td></td><td></td><td></td><td></td><td></td></tr></tbody></table></div>"`;

exports[`setting onCellAdded to invalid type should render default calendar 1`] = `"<div><table><thead><tr><th>Sun</th><th>Mon</th><th>Tue</th><th>Wed</th><th>Thu</th><th>Fri</th><th>Sat</th></tr></thead><tbody><tr><td></td><td></td><td data-date="2021-06-01">1</td><td data-date="2021-06-02">2</td><td data-date="2021-06-03">3</td><td data-date="2021-06-04">4</td><td data-date="2021-06-05">5</td></tr><tr><td data-date="2021-06-06">6</td><td data-date="2021-06-07">7</td><td data-date="2021-06-08">8</td><td data-date="2021-06-09">9</td><td data-date="2021-06-10">10</td><td data-date="2021-06-11">11</td><td data-date="2021-06-12">12</td></tr><tr><td data-date="2021-06-13">13</td><td data-date="2021-06-14">14</td><td data-date="2021-06-15">15</td><td data-date="2021-06-16">16</td><td data-date="2021-06-17">17</td><td data-date="2021-06-18">18</td><td data-date="2021-06-19">19</td></tr><tr><td data-date="2021-06-20">20</td><td data-date="2021-06-21">21</td><td data-date="2021-06-22">22</td><td data-date="2021-06-23">23</td><td data-date="2021-06-24">24</td><td data-date="2021-06-25">25</td><td data-date="2021-06-26">26</td></tr><tr><td data-date="2021-06-27">27</td><td data-date="2021-06-28">28</td><td data-date="2021-06-29">29</td><td data-date="2021-06-30">30</td><td></td><td></td><td></td></tr></tbody></table></div>"`;

exports[`setting onHeaderCellAdded to invalid type should render default calendar 1`] = `"<div><table><thead><tr><th>Sun</th><th>Mon</th><th>Tue</th><th>Wed</th><th>Thu</th><th>Fri</th><th>Sat</th></tr></thead><tbody><tr><td></td><td></td><td></td><td></td><td></td><td data-date="2024-11-01">1</td><td data-date="2024-11-02">2</td></tr><tr><td data-date="2024-11-03">3</td><td data-date="2024-11-04">4</td><td data-date="2024-11-05">5</td><td data-date="2024-11-06">6</td><td data-date="2024-11-07">7</td><td data-date="2024-11-08">8</td><td data-date="2024-11-09">9</td></tr><tr><td data-date="2024-11-10">10</td><td data-date="2024-11-11">11</td><td data-date="2024-11-12">12</td><td data-date="2024-11-13">13</td><td data-date="2024-11-14">14</td><td data-date="2024-11-15">15</td><td data-date="2024-11-16">16</td></tr><tr><td data-date="2024-11-17">17</td><td data-date="2024-11-18">18</td><td data-date="2024-11-19">19</td><td data-date="2024-11-20">20</td><td data-date="2024-11-21">21</td><td data-date="2024-11-22">22</td><td data-date="2024-11-23">23</td></tr><tr><td data-date="2024-11-24">24</td><td data-date="2024-11-25">25</td><td data-date="2024-11-26">26</td><td data-date="2024-11-27">27</td><td data-date="2024-11-28">28</td><td data-date="2024-11-29">29</td><td data-date="2024-11-30">30</td></tr></tbody></table></div>"`;

exports[`setting options.dayNames should reflect on calendar 1`] = `"<div><table><thead><tr><th>Su</th><th>Mo</th><th>Tu</th><th>We</th><th>Th</th><th>Fr</th><th>Sa</th></tr></thead><tbody><tr><td></td><td></td><td data-date="2021-06-01">1</td><td data-date="2021-06-02">2</td><td data-date="2021-06-03">3</td><td data-date="2021-06-04">4</td><td data-date="2021-06-05">5</td></tr><tr><td data-date="2021-06-06">6</td><td data-date="2021-06-07">7</td><td data-date="2021-06-08">8</td><td data-date="2021-06-09">9</td><td data-date="2021-06-10">10</td><td data-date="2021-06-11">11</td><td data-date="2021-06-12">12</td></tr><tr><td data-date="2021-06-13">13</td><td data-date="2021-06-14">14</td><td data-date="2021-06-15">15</td><td data-date="2021-06-16">16</td><td data-date="2021-06-17">17</td><td data-date="2021-06-18">18</td><td data-date="2021-06-19">19</td></tr><tr><td data-date="2021-06-20">20</td><td data-date="2021-06-21">21</td><td data-date="2021-06-22">22</td><td data-date="2021-06-23">23</td><td data-date="2021-06-24">24</td><td data-date="2021-06-25">25</td><td data-date="2021-06-26">26</td></tr><tr><td data-date="2021-06-27">27</td><td data-date="2021-06-28">28</td><td data-date="2021-06-29">29</td><td data-date="2021-06-30">30</td><td></td><td></td><td></td></tr></tbody></table></div>"`;

exports[`setting options.startDay to Fri should render a calendar that starts on a Friday 1`] = `"<div><table><thead><tr><th>Fri</th><th>Sat</th><th>Sun</th><th>Mon</th><th>Tue</th><th>Wed</th><th>Thu</th></tr></thead><tbody><tr><td></td><td></td><td></td><td></td><td data-date="2021-06-01">1</td><td data-date="2021-06-02">2</td><td data-date="2021-06-03">3</td></tr><tr><td data-date="2021-06-04">4</td><td data-date="2021-06-05">5</td><td data-date="2021-06-06">6</td><td data-date="2021-06-07">7</td><td data-date="2021-06-08">8</td><td data-date="2021-06-09">9</td><td data-date="2021-06-10">10</td></tr><tr><td data-date="2021-06-11">11</td><td data-date="2021-06-12">12</td><td data-date="2021-06-13">13</td><td data-date="2021-06-14">14</td><td data-date="2021-06-15">15</td><td data-date="2021-06-16">16</td><td data-date="2021-06-17">17</td></tr><tr><td data-date="2021-06-18">18</td><td data-date="2021-06-19">19</td><td data-date="2021-06-20">20</td><td data-date="2021-06-21">21</td><td data-date="2021-06-22">22</td><td data-date="2021-06-23">23</td><td data-date="2021-06-24">24</td></tr><tr><td data-date="2021-06-25">25</td><td data-date="2021-06-26">26</td><td data-date="2021-06-27">27</td><td data-date="2021-06-28">28</td><td data-date="2021-06-29">29</td><td data-date="2021-06-30">30</td><td></td></tr></tbody></table></div>"`;

exports[`setting options.startDay to Mon should render a calendar that starts on a Monday 1`] = `"<div><table><thead><tr><th>Mon</th><th>Tue</th><th>Wed</th><th>Thu</th><th>Fri</th><th>Sat</th><th>Sun</th></tr></thead><tbody><tr><td></td><td data-date="2021-06-01">1</td><td data-date="2021-06-02">2</td><td data-date="2021-06-03">3</td><td data-date="2021-06-04">4</td><td data-date="2021-06-05">5</td><td data-date="2021-06-06">6</td></tr><tr><td data-date="2021-06-07">7</td><td data-date="2021-06-08">8</td><td data-date="2021-06-09">9</td><td data-date="2021-06-10">10</td><td data-date="2021-06-11">11</td><td data-date="2021-06-12">12</td><td data-date="2021-06-13">13</td></tr><tr><td data-date="2021-06-14">14</td><td data-date="2021-06-15">15</td><td data-date="2021-06-16">16</td><td data-date="2021-06-17">17</td><td data-date="2021-06-18">18</td><td data-date="2021-06-19">19</td><td data-date="2021-06-20">20</td></tr><tr><td data-date="2021-06-21">21</td><td data-date="2021-06-22">22</td><td data-date="2021-06-23">23</td><td data-date="2021-06-24">24</td><td data-date="2021-06-25">25</td><td data-date="2021-06-26">26</td><td data-date="2021-06-27">27</td></tr><tr><td data-date="2021-06-28">28</td><td data-date="2021-06-29">29</td><td data-date="2021-06-30">30</td><td></td><td></td><td></td><td></td></tr></tbody></table></div>"`;

exports[`setting options.startDay to Sat should render a calendar that starts on a Saturday 1`] = `"<div><table><thead><tr><th>Sat</th><th>Sun</th><th>Mon</th><th>Tue</th><th>Wed</th><th>Thu</th><th>Fri</th></tr></thead><tbody><tr><td></td><td></td><td></td><td data-date="2021-06-01">1</td><td data-date="2021-06-02">2</td><td data-date="2021-06-03">3</td><td data-date="2021-06-04">4</td></tr><tr><td data-date="2021-06-05">5</td><td data-date="2021-06-06">6</td><td data-date="2021-06-07">7</td><td data-date="2021-06-08">8</td><td data-date="2021-06-09">9</td><td data-date="2021-06-10">10</td><td data-date="2021-06-11">11</td></tr><tr><td data-date="2021-06-12">12</td><td data-date="2021-06-13">13</td><td data-date="2021-06-14">14</td><td data-date="2021-06-15">15</td><td data-date="2021-06-16">16</td><td data-date="2021-06-17">17</td><td data-date="2021-06-18">18</td></tr><tr><td data-date="2021-06-19">19</td><td data-date="2021-06-20">20</td><td data-date="2021-06-21">21</td><td data-date="2021-06-22">22</td><td data-date="2021-06-23">23</td><td data-date="2021-06-24">24</td><td data-date="2021-06-25">25</td></tr><tr><td data-date="2021-06-26">26</td><td data-date="2021-06-27">27</td><td data-date="2021-06-28">28</td><td data-date="2021-06-29">29</td><td data-date="2021-06-30">30</td><td></td><td></td></tr></tbody></table></div>"`;

exports[`setting options.startDay to Sun should render a calendar that starts on a Sunday 1`] = `"<div><table><thead><tr><th>Sun</th><th>Mon</th><th>Tue</th><th>Wed</th><th>Thu</th><th>Fri</th><th>Sat</th></tr></thead><tbody><tr><td></td><td></td><td data-date="2021-06-01">1</td><td data-date="2021-06-02">2</td><td data-date="2021-06-03">3</td><td data-date="2021-06-04">4</td><td data-date="2021-06-05">5</td></tr><tr><td data-date="2021-06-06">6</td><td data-date="2021-06-07">7</td><td data-date="2021-06-08">8</td><td data-date="2021-06-09">9</td><td data-date="2021-06-10">10</td><td data-date="2021-06-11">11</td><td data-date="2021-06-12">12</td></tr><tr><td data-date="2021-06-13">13</td><td data-date="2021-06-14">14</td><td data-date="2021-06-15">15</td><td data-date="2021-06-16">16</td><td data-date="2021-06-17">17</td><td data-date="2021-06-18">18</td><td data-date="2021-06-19">19</td></tr><tr><td data-date="2021-06-20">20</td><td data-date="2021-06-21">21</td><td data-date="2021-06-22">22</td><td data-date="2021-06-23">23</td><td data-date="2021-06-24">24</td><td data-date="2021-06-25">25</td><td data-date="2021-06-26">26</td></tr><tr><td data-date="2021-06-27">27</td><td data-date="2021-06-28">28</td><td data-date="2021-06-29">29</td><td data-date="2021-06-30">30</td><td></td><td></td><td></td></tr></tbody></table></div>"`;

exports[`setting options.startDay to Thu should render a calendar that starts on a Thursday 1`] = `"<div><table><thead><tr><th>Thu</th><th>Fri</th><th>Sat</th><th>Sun</th><th>Mon</th><th>Tue</th><th>Wed</th></tr></thead><tbody><tr><td></td><td></td><td></td><td></td><td></td><td data-date="2021-06-01">1</td><td data-date="2021-06-02">2</td></tr><tr><td data-date="2021-06-03">3</td><td data-date="2021-06-04">4</td><td data-date="2021-06-05">5</td><td data-date="2021-06-06">6</td><td data-date="2021-06-07">7</td><td data-date="2021-06-08">8</td><td data-date="2021-06-09">9</td></tr><tr><td data-date="2021-06-10">10</td><td data-date="2021-06-11">11</td><td data-date="2021-06-12">12</td><td data-date="2021-06-13">13</td><td data-date="2021-06-14">14</td><td data-date="2021-06-15">15</td><td data-date="2021-06-16">16</td></tr><tr><td data-date="2021-06-17">17</td><td data-date="2021-06-18">18</td><td data-date="2021-06-19">19</td><td data-date="2021-06-20">20</td><td data-date="2021-06-21">21</td><td data-date="2021-06-22">22</td><td data-date="2021-06-23">23</td></tr><tr><td data-date="2021-06-24">24</td><td data-date="2021-06-25">25</td><td data-date="2021-06-26">26</td><td data-date="2021-06-27">27</td><td data-date="2021-06-28">28</td><td data-date="2021-06-29">29</td><td data-date="2021-06-30">30</td></tr></tbody></table></div>"`;

exports[`setting options.startDay to Tue should render a calendar that starts on a Tuesday 1`] = `"<div><table><thead><tr><th>Tue</th><th>Wed</th><th>Thu</th><th>Fri</th><th>Sat</th><th>Sun</th><th>Mon</th></tr></thead><tbody><tr><td data-date="2021-06-01">1</td><td data-date="2021-06-02">2</td><td data-date="2021-06-03">3</td><td data-date="2021-06-04">4</td><td data-date="2021-06-05">5</td><td data-date="2021-06-06">6</td><td data-date="2021-06-07">7</td></tr><tr><td data-date="2021-06-08">8</td><td data-date="2021-06-09">9</td><td data-date="2021-06-10">10</td><td data-date="2021-06-11">11</td><td data-date="2021-06-12">12</td><td data-date="2021-06-13">13</td><td data-date="2021-06-14">14</td></tr><tr><td data-date="2021-06-15">15</td><td data-date="2021-06-16">16</td><td data-date="2021-06-17">17</td><td data-date="2021-06-18">18</td><td data-date="2021-06-19">19</td><td data-date="2021-06-20">20</td><td data-date="2021-06-21">21</td></tr><tr><td data-date="2021-06-22">22</td><td data-date="2021-06-23">23</td><td data-date="2021-06-24">24</td><td data-date="2021-06-25">25</td><td data-date="2021-06-26">26</td><td data-date="2021-06-27">27</td><td data-date="2021-06-28">28</td></tr><tr><td data-date="2021-06-29">29</td><td data-date="2021-06-30">30</td><td></td><td></td><td></td><td></td><td></td></tr></tbody></table></div>"`;

exports[`setting options.startDay to Wed should render a calendar that starts on a Wednesday 1`] = `"<div><table><thead><tr><th>Wed</th><th>Thu</th><th>Fri</th><th>Sat</th><th>Sun</th><th>Mon</th><th>Tue</th></tr></thead><tbody><tr><td></td><td></td><td></td><td></td><td></td><td></td><td data-date="2021-06-01">1</td></tr><tr><td data-date="2021-06-02">2</td><td data-date="2021-06-03">3</td><td data-date="2021-06-04">4</td><td data-date="2021-06-05">5</td><td data-date="2021-06-06">6</td><td data-date="2021-06-07">7</td><td data-date="2021-06-08">8</td></tr><tr><td data-date="2021-06-09">9</td><td data-date="2021-06-10">10</td><td data-date="2021-06-11">11</td><td data-date="2021-06-12">12</td><td data-date="2021-06-13">13</td><td data-date="2021-06-14">14</td><td data-date="2021-06-15">15</td></tr><tr><td data-date="2021-06-16">16</td><td data-date="2021-06-17">17</td><td data-date="2021-06-18">18</td><td data-date="2021-06-19">19</td><td data-date="2021-06-20">20</td><td data-date="2021-06-21">21</td><td data-date="2021-06-22">22</td></tr><tr><td data-date="2021-06-23">23</td><td data-date="2021-06-24">24</td><td data-date="2021-06-25">25</td><td data-date="2021-06-26">26</td><td data-date="2021-06-27">27</td><td data-date="2021-06-28">28</td><td data-date="2021-06-29">29</td></tr><tr><td data-date="2021-06-30">30</td><td></td><td></td><td></td><td></td><td></td><td></td></tr></tbody></table></div>"`;
